htlcswitch/mock.go

```diff
@@ -183,8 +183,8 @@ func newMockServer(t testing.TB, name string, startingHeight uint32,
 	db *channeldb.DB, defaultDelta uint32) (*mockServer, error) {
 
 	var id [33]byte
-	h := sha256.Sum256([]byte(name))
-	copy(id[:], h[:])
```

Comment:
Previously it was generating an invalid key which was throwing errors (invalid magic, square root) on some tests with the addition of `targetPeer.IdentityKey().SerializeCompressed()` at `htlcswitch/switch.go`.
So I changed it to generate the key using btcec using the provided name bytes, probably slower than before tho.

Reply:
Is this change still needed?

Reply:
Tested and looks fine, tests also fine. Removed, thx.